Source/Core/DolphinQt/GameList/GameList.cpp

```diff
@@ -676,7 +705,14 @@ void GameList::DeleteFile()
 
 void GameList::ChangeDisc()
 {
-  Core::RunAsCPUThread([this] { DVDInterface::ChangeDisc(GetSelectedGame()->GetFilePath()); });
+  const auto game = GetSelectedGame();
+  if (!game)
+    return;
+
+  Core::RunAsCPUThread([game]
```

Comment:
Is it really a good idea to capture a `std::shared_ptr` in a lambda expression? It kinda raised an eyebrow, and google came up with [some](https://floating.io/2017/07/lambda-shared_ptr-memory-leak/) [results](https://pusling.com/blog/?p=446) that either straight-out say "don't"  or at least "think VERY carefully before doing this".

Not sure what the better option would be, though. Capturing `this`, and calling `GetSelectedGame()` inside the lambda, doing the check there instead of outside?

Reply:
You should be able to capture the string that `GetFilePath()` returns.

Reply:
FYI, there's another capture in `GameList::ShowContextMenu` further up top (calling `WiiUpdate::PerformDiscUpdate`, which could use the same technique)

Reply:
As suggested, now only the file path is being captured in both cases.